pkg/deploy/interfaces.go

```diff
@@ -8,13 +8,13 @@ import (
 )
 
 type Deploy interface {
-	Deploy(tplOutput string, startedApplyingFunc func(),
+	Deploy(tplOutput string, appName string, startedApplyingFunc func(),
```

Comment:
i think it would make sense to maintain Deploy interface as agnostic to appname. can this be injected into the constructor?

Reply:
@cppforlife I opted for in-place editing of the appName ==> see https://github.com/vmware-tanzu/carvel-kapp-controller/pull/665/files#diff-33a2bb066bb47c93866134d1ee796e9bcad9ed87d6248e10428c21dd1183d245R34

WDYT? I'm not a huge fan of mutating state. 

We can change the `newKapp` arguments to accept a different name, and use that (still overwriting the name in `GenericOpts`). Or the alternative is a new key in our constructor. I like the idea of the calling functions having the power over the name, so I went with this edit-in-place model for now.

Reply:
> WDYT? I'm not a huge fan of mutating state though, which this kinda does.

agreed, mutation is not great. here is what im thinking of:

- NewKapp constructor can take a "suffix": https://github.com/vmware-tanzu/carvel-kapp-controller/blob/58ce5c8bd33e9b930e0d58a5299b2d563b24c0dc/pkg/deploy/kapp.go#L37. deploy.Factory.NewKapp method can instantiate NewKapp with ".app" suffix: https://github.com/vmware-tanzu/carvel-kapp-controller/blob/58ce5c8bd33e9b930e0d58a5299b2d563b24c0dc/pkg/deploy/factory.go#L64

- deploy.Factory.NewKappPrivileged can be renamed to NewKappPrivelegedForPackageRepository which would construct NewKapp with ".pkgr" suffix. https://github.com/vmware-tanzu/carvel-kapp-controller/blob/58ce5c8bd33e9b930e0d58a5299b2d563b24c0dc/pkg/deploy/factory.go#L70

this way this stuff is hidden from the rest of the code and the interface to deployment remains the same.

Reply:
Cool, I actually had it this way at some point. I ended up changing because I didn't want to make the `newKapp` vs `NewKappPrivileged` encode the `.app` vs `.pkgr` differences. But sounds like `NewKappPrivileged` will only EVER be used by package repositories, so this assumption is fine now. Thanks for the clarity